dll/win32/shell32/dialogs/filetypes.cpp

```diff
@@ -754,21 +764,13 @@ FileTypesDlg_InsertToLV(HWND hListView, LPCWSTR Assoc, INT iItem, PFILE_TYPE_GLO
         }
 #else
         if (!Entry->ClassKey[0])
-        {
-            HeapFree(pG->hHeap, 0, Entry);
-            RegCloseKey(hKey);
-            return NULL;
-        }
+            goto exclude;
```

Comment:
Slightly sad that we use `goto` in c++ code where `RAII` could handle all this stuff.

Reply:
I hear you. The next time this code is touched, it should be to do the enumeration on a background thread and then enough has to change to justify changing the allocation scheme.